internal/infrastructure/token/encrypt.go

```diff
@@ -0,0 +1,67 @@
+package token
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/json"
+	"errors"
+	"io"
+
+	"github.com/lestrrat-go/jwx/v2/jwk"
+)
+
+// Encrypt a JWK Key with encryption key
+// Uses AES-GCM with a random nonce. The nonce is pre-pended to ciphertext.
+func EncryptJWK(aeskey []byte, jwkey jwk.Key) (enc []byte, err error) {
+	// create AES-GCM AEAD and nonce
+	block, err := aes.NewCipher(aeskey)
+	if err != nil {
+		return nil, err
+	}
+	aesgcm, err := cipher.NewGCM(block)
+	if err != nil {
+		return nil, err
+	}
+	nonce := make([]byte, aesgcm.NonceSize())
+	if _, err := io.ReadFull(rand.Reader, nonce); err != nil {
+		return nil, err
+	}
+
+	// serialize JWK to JSON bytes
+	jwkbuf, err := json.Marshal(jwkey)
+	if err != nil {
+		return nil, err
+	}
+
+	// encrypt and seal
+	ciphertext := aesgcm.Seal(nil, nonce, jwkbuf, nil)
```

Comment:
This Go crypto API is a bit odd, and not adequately documented.  From reading the source code, I *think* the follow would work too.  (Not that I am asking for this change; I prefer it explicit, the way you wrote it):

```go
    return aesgcm.Seal(nonce, nonce, jwkbuf, nil)
```

The first argument, `dst`, will be copied to the output.  In fact, if `cap(dst) >= len(dst) + len(plaintext) + tagSize` it will be used as-is, and the ciphertext and tag will simply be appended without making a new buffer.  So if you make the nonce buffer big enough and only read `NonceSize()` random bytes into it, there will only be one new buffer allocated for the whole encryption operation and the output is exactly what we want.

Again all of this is overkill for us, and harms readability.  I just want to share what I have learned about this API :)